lib/binary_to_decimal.rb

```diff
@@ -4,6 +4,17 @@
 # The least significant bit is at index 7.
 # Calculate  and return the decimal value for this binary number using
 # the algorithm you devised in class.
+# def binary_to_decimal(binary_array)
+#   raise NotImplementedError
+# end
+
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  decimal_value = 0
+  binary_array.each_with_index do |bit, i|
+    if (bit == 1)
+      decimal_value += (2 ** ((binary_array.length - 1) - i))
+    end
```

Comment:
You could simplify this a bit, and just multiply by `bit`.  Since if it's a 1, you add the correct power of 2 and if it's a zero you just add 0 to `decimal_value`.

```suggestion
    decimal_value += (2 ** ((binary_array.length - 1) - i)) * bit
```

Reply:
Love that, thanks, Chris!